cmd/generator/main.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2020 GramLabs, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// wtf. https://github.com/golang/tools/blob/master/go/analysis/passes/buildtag/buildtag.go#L47-L51
+// // +build ignore
+
+// Generator handles embedding non-go files into the compiled binary.
+// Embedded assets are gzipped and base64 encoded.
+// This makes use of redskyctl/internal/kustomize/assets.go to interact with the
+// embedded data.
+package main
+
+import (
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"strings"
+
+	flag "github.com/spf13/pflag"
+	"sigs.k8s.io/kustomize/api/filesys"
+	"sigs.k8s.io/kustomize/api/krusty"
+)
+
+func main() {
+	var (
+		path          = flag.String("path", ".", "path of kustomization.yaml")
+		outputPackage = flag.String("package", "generated", "output package name")
```

Comment:
This isn't configurable because of the reference to the `Asset` struct; you would need to conditionally add an `import` statement to keep it configurable.

Reply:
There's a dependency on having an Asset type defined, but the package name itself doesn't matter ( we're generating it for `package kustomize` ). We could probably move the `redskyctl/internal/kustomize/assets.go` bit into this command to generate the assets struct as well, but I'm not sure it's worth tossing all of that go code inside of a string literal or the like.